src/everest/config/simulator_config.py

```diff
@@ -58,6 +59,29 @@ class SimulatorConfig(BaseModelWithContextSupport, extra="forbid"):
         A value of 0 means unlimited runtime.
         """,
     )
+    max_memory: int | str | None = Field(
```

Comment:
This would entail a bigger refactor, which would also touch ert, but there is a concept in ert of generic queue options, and I think this one probably belongs with that https://github.com/equinor/ert/blob/99d7804e1552a26070e0a592f56a1e76c71ce2d0/src/ert/config/queue_config.py#L40

Reply:
Yes, I noted that also. It seems not everything lives in there though, for instance the `QueueOptions` base class does contain `max_running`, but not `max_runtime`, or `realization_memory`, they live in `QueueConfig`. Maybe `realization_memory` does belong in the options class since it is basically a property you set in the node. But maybe that is true for some of the other fields too? Changing that should maybe a separate issue since it touches ERT.

Also, then the question is if it changes the Everest config: Do we keep `max_memory` as a field in `SimulatorConfig`? Or does the user have to specify it in the queue-specific options?

Reply:
This is somewhat related to this issue https://github.com/equinor/ert/issues/10628. If we had the same keyword for max memory for everest and ert it would simplify giving the user error messages like this.

Reply:
> Also, then the question is if it changes the Everest config: Do we keep max_memory as a field in SimulatorConfig? Or does the user have to specify it in the queue-specific options?

Think it belongs in the queue options, as it is directly translated to the driver. If so we could set it there as the desired keyword in this PR, and then do the refactor to generic queue option in another follow up PR. That way we dont have to break the config more than once.

Then we can also align it with the ert name right away.

Reply:
I am not quite sure what to do about this. Currently, `realization_memory` lives in `QueueConfig`, moving it to `QueueOptions` to make it a queue option would have consequences for the ERT configuration, right? Not sure how to handler that. We could add a `max_memory` to `QueueOptions` for the benefit of Everest, but would imply some duplicated functionality and it would not align well with ERT. How shall we proceed?

Reply:
As discussed, lets create a follow up issue for this 👍